lib/elixir/lib/task/supervised.ex

```diff
@@ -211,26 +215,43 @@ defmodule Task.Supervised do
           case waiting do
             %{^position => {_, {:ok, _} = ok}} -> Map.put(waiting, position, {nil, ok})
             %{^position => {_, :running}} -> Map.put(waiting, position, {nil, {:exit, reason}})
+            %{^position => {_, :timed_out}} -> Map.put(waiting, position, {nil, {:exit, :killed}})
```

Comment:
We can not guarantee the reason is `:killed` here, it can be any reason. For example task can exit while timer message is in the message queue of the Monitor. Since it would be `:killed` if it was killed we could just use the reason in the `:exit`. Also the reason might be `:killed` because the task was killed for another cause.

Therefore I think we should use another value when `:timed_out` and reason is `:killed`, otherwise do `{:exit, reason}`.